addons/website_blog/models/website_blog.py

```diff
@@ -334,14 +312,14 @@ def _search_get_detail(self, website, order, options):
             if active_tag_ids:
                 domain.append([('tag_ids', 'in', active_tag_ids)])
         if date_begin and date_end:
-            domain.append([("post_date", ">=", date_begin), ("post_date", "<=", date_end)])
+            domain.append([("published_date", ">=", date_begin), ("published_date", "<=", date_end)])
         if self.env.user.has_group('website.group_website_designer'):
             if state == "published":
-                domain.append([("website_published", "=", True), ("post_date", "<=", fields.Datetime.now())])
+                domain.append([("website_published", "=", True)])
             elif state == "unpublished":
-                domain.append(['|', ("website_published", "=", False), ("post_date", ">", fields.Datetime.now())])
+                domain.append([("website_published", "=", False)])
```

Comment:
Why don't you check `is_published` here? (Just like in `addons/website_blog/controllers/main.py`.)

Reply:
@robinlej As a reminder after we talk, website_published is a related field for is_published in the "WebsitePublishedMixin" and isn't related in the "WebsitePublishedMultiMixin". So we have to use "website_published" when defined.